certbot/certbot/plugins/common.py

```diff
@@ -39,13 +40,10 @@ def dest_namespace(name):
     r"^(([a-z0-9]|[a-z0-9][a-z0-9\-]*[a-z0-9])\.)*[a-z]+$", re.IGNORECASE)
 
 
-@zope.interface.implementer(interfaces.IPlugin)
-class Plugin:
+class Plugin(AbstractPlugin, metaclass=ABCMeta):  # pylint: disable=abstract-method
     """Generic plugin."""
-    # provider is not inherited, subclasses must define it on their own
-    # @zope.interface.provider(interfaces.IPluginFactory)
 
-    def __init__(self, config, name):
+    def __init__(self, config, name):  # pylint: disable=super-init-not-called
```

Comment:
Option 1. feels the more natural. Previously pylint was complaining either of a useless call of super or no call of super when the logic of the constructor was implemented on the interface... But now it is fine since the logic is on the class.

Reply:
This is a response to https://github.com/certbot/certbot/pull/8950#discussion_r677809761.

I actually was proposing we do both (1) and (2) and also call `super().__init__()` in the plugin interface. My reasoning behind was to ensure the class works as expected when multiple inheritance is used.

On second thought, I think this is a little tricky though because it's not immediately clear to me whether we should call `super().__init__()` or `super().__init__(config, name)` here. If you have any thoughts about any of this, I'd be interested to hear them, but I think this is enough of an edge case that I'm not worried about it anymore.

Reply:
Ah yes, of course! I used `super().__init__()` on the interface level since I do not think there is any specific logic to say about the super constructor here. The provided arguments to the constructor are here for the contract purpose, but there is no specific logic tied to the interface.